crates/nu-protocol/src/pipeline/byte_stream.rs

```diff
@@ -114,6 +114,14 @@ impl ByteStreamType {
             ByteStreamType::Unknown => "byte stream",
         }
     }
+
+    pub fn maybe_binary(self) -> bool {
+        matches!(self, ByteStreamType::Binary | ByteStreamType::Unknown)
+    }
+
+    pub fn maybe_string(self) -> bool {
+        matches!(self, ByteStreamType::String | ByteStreamType::Unknown)
+    }
```

Comment:
Not totally sure how to name these methods.

Reply:
Suggestions:

- `is_binary_compatible()`
- `is_binary_like()`
- `is_like_binary()`